source/gui/addonGui.py

```diff
@@ -297,29 +548,191 @@ def onEnableDisable(self, evt):
 		if index<0: return
 		addon=self.curAddons[index]
 		shouldDisable = self._shouldDisable(addon)
-		# Counterintuitive, but makes sense when context is taken into account.
-		addon.enable(not shouldDisable)
+		try:
+			# Counterintuitive, but makes sense when context is taken into account.
+			addon.enable(not shouldDisable)
+		except addonHandler.AddonError:
+			log.error("Couldn't change state for %s add-on"%addon.name, exc_info=True)
+			gui.messageBox(
+				# Translators: The message displayed when the state of an add-on cannot be changed.

```

Comment:
May be we should create separate complete messages for enable and disable, because otherwise some languages might get into grammatical issues.